worker/uniter/relation/relations.go

```diff
@@ -112,36 +113,47 @@ func NewRelations(st *uniter.State, tag names.UnitTag, charmDir, relationsDir st
 // the corresponding relations. It's only expected to be called while a
 // *relations is being created.
 func (r *relations) init() error {
-	joinedRelationTags, err := r.unit.JoinedRelations()
+	relationStatus, err := r.unit.RelationsStatus()
 	if err != nil {
 		return errors.Trace(err)
 	}
 	// Keep the relations ordered for reliable testing.
 	var orderedIds []int
-	joinedRelations := make(map[int]*uniter.Relation)
-	for _, tag := range joinedRelationTags {
-		relation, err := r.st.Relation(tag)
+	activeRelations := make(map[int]*uniter.Relation)
+	relationStatusValues := make(map[int]relation.Status)
+	for _, rs := range relationStatus {
+		if !rs.InScope {
+			continue
+		}
+		relation, err := r.st.Relation(rs.Tag)
 		if err != nil {
 			return errors.Trace(err)
 		}
-		joinedRelations[relation.Id()] = relation
+		relationStatusValues[relation.Id()] = rs.Status
+		activeRelations[relation.Id()] = relation
 		orderedIds = append(orderedIds, relation.Id())
 	}
 	knownDirs, err := ReadAllStateDirs(r.relationsDir)
 	if err != nil {
 		return errors.Trace(err)
 	}
 	for id, dir := range knownDirs {
-		if rel, ok := joinedRelations[id]; ok {
+		if rel, ok := activeRelations[id]; ok {
 			if err := r.add(rel, dir); err != nil {
 				return errors.Trace(err)
 			}
-		} else if err := dir.Remove(); err != nil {
-			return errors.Trace(err)
+		} else {
+			switch relationStatusValues[id] {
+			case relation.Joined, relation.Suspended, relation.Error:
```

Comment:
comment to the effect that these relations might later become active again

Reply:
ok